src/goto-instrument/contracts/contracts.cpp

```diff
@@ -598,22 +668,55 @@ bool code_contractst::apply_function_contract(
 }
 
 void code_contractst::apply_loop_contract(
-  const irep_idt &function,
+  const irep_idt &function_name,
   goto_functionst::goto_functiont &goto_function)
 {
   local_may_aliast local_may_alias(goto_function);
   natural_loops_mutablet natural_loops(goto_function.body);
 
-  // Iterate over the (natural) loops in the function,
-  // and apply any invariant annotations that we find.
+  // a graph node type that stores information about a loop
```

Comment:
```suggestion
  // A graph node type that stores information about a loop.
```

We could add a couple more comments here. For instance, which information are you referring to (e.g., loop body only, or an entire copy of the loop instructions, or only the set of local variables, or just loop head...)? Why do we need to use a graph to store this information and not another standard data structure instead? I see that every node contains "loop.first" and "loop.second", which I assume correspond to `loop_head` and `loop_body`, respectively - am I right? What about target?

I know that most of this information I could probably deduce from debugging the code a bit more. I also know that having these loops in topological order is the reason why we create this graph (unless I'm wrong). However, it might improve maintainability if we add a couple of more comments to make these bits of information even more obvious, specially answering questions like "_why topological order is important here?_" and set developers in the right direction next time they scan this code.